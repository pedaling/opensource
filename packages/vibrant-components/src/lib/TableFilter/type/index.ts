export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
