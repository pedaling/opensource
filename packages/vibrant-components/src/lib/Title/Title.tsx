import { Text } from '@vibrant-ui/core';
import { withTitleVariation } from './TitleProps';

export const Title = withTitleVariation(({ innerRef, ...restProps }) => <Text ref={innerRef} {...restProps} />);
