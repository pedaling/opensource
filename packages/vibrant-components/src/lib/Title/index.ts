export { Title } from './Title';
export type { TitleProps } from './TitleProps';
