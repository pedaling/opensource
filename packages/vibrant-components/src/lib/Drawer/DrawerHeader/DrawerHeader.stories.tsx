import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { DrawerHeader } from './DrawerHeader';

export default {
  title: 'DrawerHeader',
  component: DrawerHeader,
  args: {},
} as ComponentMeta<typeof DrawerHeader>;

export const Basic: ComponentStory<typeof DrawerHeader> = props => <DrawerHeader {...props} />;
