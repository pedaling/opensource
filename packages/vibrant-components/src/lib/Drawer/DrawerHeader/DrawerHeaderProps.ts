import type { ReactElement } from 'react';
import { withVariation } from '@vibrant-ui/core';

export type DrawerHeaderProps = {
  testId?: string;
  closable?: boolean;
} & (
  | {
      children: ReactElement | ReactElement[];
      title?: never;
    }
  | {
      children?: never;
      title: string;
    }
);

export const withDrawerHeaderVariation = withVariation<DrawerHeaderProps>('DrawerHeader')();
