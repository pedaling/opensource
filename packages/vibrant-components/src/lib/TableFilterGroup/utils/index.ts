import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
