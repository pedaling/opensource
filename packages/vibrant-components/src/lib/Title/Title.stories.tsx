import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Title } from './Title';

export default {
  title: 'Title',
  component: Title,
  args: {
    level: 1,
    children: 'Sample Text',
  },
} as ComponentMeta<typeof Title>;

export const Basic: ComponentStory<typeof Title> = ({ children, ...restProps }) => (
  <Title {...restProps}>{children}</Title>
);
