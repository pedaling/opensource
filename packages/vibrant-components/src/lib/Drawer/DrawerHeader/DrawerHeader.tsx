import { Box, Text } from '@vibrant-ui/core';
import { Icon } from '@vibrant-ui/icons';
import { Divider } from '../../Divider';
import { HStack } from '../../HStack';
import { IconButton } from '../../IconButton';
import { useDrawer } from '../DrawerContext';
import { withDrawerHeaderVariation } from './DrawerHeaderProps';

export const DrawerHeader = withDrawerHeaderVariation(
  ({ testId = 'drawer-header', children, title, closable = false }) => {
    const { togglePanel } = useDrawer();

    return (
      <>
        <Box data-testid={testId} py={12} px={20}>
          <HStack alignHorizontal="space-between">
            {children ? (
              children
            ) : (
              <Text
                as="h2"
                fontWeight="bold"
                typography="title6"
                color="onView1"
                wordWrap="break-word"
                wordBreak="break-word"
                whiteSpace="pre-wrap"
              >
                {title}
              </Text>
            )}
            {closable && (
              <Box py={2}>
                <IconButton
                  IconComponent={Icon.Close.Thin}
                  size="sm"
                  color="onView1"
                  onClick={togglePanel}
                  ariaLabel="close-button"
                />
              </Box>
            )}
          </HStack>
        </Box>
        <Divider direction="horizontal" kind="default" />
      </>
    );
  }
);
