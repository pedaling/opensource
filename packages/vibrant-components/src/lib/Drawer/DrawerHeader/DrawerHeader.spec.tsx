import { Box } from '@vibrant-ui/core';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Title } from '../../Title';
import { DrawerHeader } from './DrawerHeader';

describe('<DrawerHeader />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when title is activated', () => {
    beforeEach(async () => {
      renderer = render(<DrawerHeader title="DrawerHeader test" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when children is set', () => {
    beforeEach(async () => {
      renderer = render(
        <DrawerHeader closable={true}>
          <Box display="flex" flexDirection="row" justifyContent="flex-start" alignItems="stretch">
            <Title level={3}>DrawerHeader custom title</Title>
            <Title level={3}>DrawerHeader custom subtitle</Title>
          </Box>
        </DrawerHeader>
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when closable is true', () => {
    beforeEach(async () => {
      renderer = render(<DrawerHeader title="DrawerHeader test" closable={true} />);
    });

    it('should render close button', () => {
      expect(renderer.getByRole('button', { name: 'close-button' })).toBeTruthy();
    });
  });
});
