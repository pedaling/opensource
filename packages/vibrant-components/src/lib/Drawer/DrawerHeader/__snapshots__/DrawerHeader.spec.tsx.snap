// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerHeader /> when children is set match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 20px;
  padding-right: 20px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 1.625rem;
  line-height: 2.25rem;
  font-weight: 700;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 2px;
  padding-bottom: 2px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  border-radius: 2px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 16px;
  height: 16px;
  fill: #0c0c0c;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer-header"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <h2
          class="emotion-3"
          data-testid="title"
        >
          DrawerHeader custom title
        </h2>
        <h2
          class="emotion-3"
          data-testid="title"
        >
          DrawerHeader custom subtitle
        </h2>
      </div>
      <div
        class="emotion-5"
      >
        <button
          aria-label="닫기"
          class="emotion-6"
          data-testid="icon-button"
          style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
          type="button"
        >
          <svg
            class="emotion-7"
            data-testid="close-thin"
            viewBox="0 0 24 24"
          >
            <path
              class="emotion-8"
              d="m13.25 12 6.6-6.6c.1-.1.1-.25 0-.35l-.85-.9c-.1-.1-.25-.1-.35 0L12 10.75l-6.6-6.6c-.1-.1-.25-.1-.35 0l-.9.85c-.1.1-.1.25 0 .35l6.6 6.65-6.6 6.6c-.1.1-.1.25 0 .35l.9.9c.1.1.25.1.35 0l6.6-6.6 6.6 6.6c.1.1.25.1.35 0l.9-.9c.1-.1.1-.25 0-.35l-6.6-6.6Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class="emotion-9"
    kind="default"
  />
</div>
`;

exports[`<DrawerHeader /> when title is activated match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 20px;
  padding-right: 20px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-2 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.25rem;
  word-wrap: break-word;
  word-break: break-word;
  white-space: pre-wrap;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

<div>
  <div
    class="emotion-0"
    data-testid="drawer-header"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-2"
      >
        DrawerHeader test
      </h2>
    </div>
  </div>
  <div
    class="emotion-3"
    kind="default"
  />
</div>
`;
